IGNITE-25847 Spring Data documentation (#6207)
apache/ignite-3: docs/_docs/developers-guide/spring-data/spring-data.adoc

The easiest way to start working with Apache Ignite's Spring Data repository is by adding the following Maven dependencies
to the application's `pom.xml` file:

[tabs]
--
tab:pom.xml[]
[source,xml]
----
<dependency>
    <groupId>org.apache.ignite</groupId>
    <artifactId>ignite-spring-ignite</artifactId>
    <version>${ignite.version}</version>
</dependency>

<dependency>
    <groupId>org.apache.ignite</groupId>
    <artifactId>spring-boot-ignite-client-autoconfigure</artifactId>
    <version>${ignite.version}</version>
</dependency>

----
--

Replace `${ignite.version}` with an actual version of Apache Ignite dependencies you are interested in, respectively.

== Application Configuration

You should provide the following properties to the Spring application:

ignite.client.addresses=${ignite.address}
spring.datasource.url=${ignite.jdbc.connectionString}
spring.datasource.driver-class-name=org.apache.ignite.jdbc.IgniteJdbcDriver

Replace `${ignite.address}` with ignite host and port. For example, `localhost:10800`. Replace ${ignite.jdbc.connectionString} with actual
JDBC connection string, for example `jdbc:ignite:thin://localhost`.

Example of application.properties file:

> ignite.client.addresses=127.0.0.1:10800

Reviewer: ```suggestion
[source, text]
----
ignite.client.addresses=127.0.0.1:10800
```

spring.datasource.url=jdbc:ignite:thin://localhost
spring.datasource.driver-class-name=org.apache.ignite.jdbc.IgniteJdbcDriver

> 

Reviewer: ```suggestion
----

```

> You should also provide `IgniteDialectProvider` by adding the following config to the `resources/META-INF/spring.factories` (create the

Reviewer: ```suggestion
You should also provide the `IgniteDialectProvider` parameter by adding the following configuration property to the `resources/META-INF/spring.factories` (create the
```

> file if it doesn't exist):

Reviewer: ```suggestion
file if it does not exist):
```

> org.springframework.data.jdbc.repository.config.DialectResolver$JdbcDialectProvider=org.apache.ignite.data.IgniteDialectProvider

Reviewer: ```suggestion
[source, text]
----
org.springframework.data.jdbc.repository.config.DialectResolver$JdbcDialectProvider=org.apache.ignite.data.IgniteDialectProvider
----
```

== Repository

> After the above configuration, you can create spring repositories.

Reviewer: Needs a verb. Maybe
After providing the above configuration...

For instance, let's create the first custom repository named `PersonRepository`:

[tabs]
--
tab:Java[]
[source,java]
----
@Repository
public interface PersonRepository extends IgniteRepository<Person, Long> {
    /**
     * Gets all the persons with the given name.
     * @param name Person name.
     * @return A list of Persons with the given first name.
     */
    public List<Person> findByFirstName(String name);

    /**
     * Returns top Person with the specified surname.
     * @param name Person surname.
     * @return Person that satisfy the query.
     */
    public Person findTopByLastNameLike(String name);

    /**
     * Getting ids of all the Person satisfying the custom query from {@link Query} annotation.
     *
     * @param orgId Query parameter.
     * @param pageable Pageable interface.
     * @return A list of Persons' ids.
     */
    @Query("SELECT id FROM Person WHERE orgId > ?")
    public List<Long> selectId(long orgId, Pageable pageable);
}
----
--

> * Signatures of custom methods like `findByFirstName(name)` and `findTopByLastNameLike(name)` will be automatically processed and turned

Reviewer: ```suggestion
Signatures of custom methods like `findByFirstName(name)` and `findTopByLastNameLike(name)` will be automatically processed and turned
```

into SQL queries when methods get executed. In addition, `@Query(queryString)` annotation can be used if a concrete​ SQL
query needs to be executed as a result of a method call.

[CAUTION]
====
[discrete]
A table must be created in Apache Ignite before working with it.

> 

Reviewer: ```suggestion
```

database to another with minimal efforts. Apache Ignite supports Spring Data JDBC by implementing `IgniteDialect`.

== Maven Configuration

The easiest way to start working with Apache Ignite's Spring Data repository is by adding the following Maven dependencies
to the application's `pom.xml` file:

[tabs]
--
tab:pom.xml[]
[source,xml, subs="attributes,specialchars"]]
----
<dependency>
    <groupId>org.apache.ignite</groupId>
    <artifactId>ignite-spring-ignite</artifactId>
    <version>${version}</version>
</dependency>

<dependency>
    <groupId>org.apache.ignite</groupId>
    <artifactId>spring-boot-ignite-client-autoconfigure</artifactId>
    <version>${version}</version>
</dependency>

----
--

== Application Configuration

To start working with Spring Data, first provide the following properties to your application:

[source, text]
----
ignite.client.addresses=${ignite.address}
spring.datasource.url=${ignite.jdbc.connectionString}
spring.datasource.driver-class-name=org.apache.ignite.jdbc.IgniteJdbcDriver
----

In the example above:

* Replace `${ignite.address}` with the host and port of your node. For example, `localhost:10800`. 

> Replace ${ignite.jdbc.connectionString} with your JDBC connection string, for example `jdbc:ignite:thin://localhost`.

Reviewer: Should be a list item
```suggestion
* Replace ${ignite.jdbc.connectionString} with your JDBC connection string, for example `jdbc:ignite:thin://localhost`.
```
